packages/binding-coap/src/coap-server.ts

```diff
@@ -344,6 +346,9 @@ export default class CoapServer implements ProtocolServer {
                                             res.write(content.body);
 
                                             res.on("finish", function (err: Error) {
+                                                if (err) {
+                                                    console.error(err);
```

Comment:
For logging we need to use our convention, I am wondering if we can enforce it somehow with eslint 🤔

Reply:
What do you mean by "our convention"?
I hope you've understood that I've added that code because of this rule - https://eslint.org/docs/rules/handle-callback-err.

Reply:
If you take a look at our packages and the console logs you will notice that we add a prefix to the error so that we can easily  detect from which binding the error came from...

e.g., 

https://github.com/danielpeintner/thingweb.node-wot/blob/59fe3264bb4cfa19a3c8cddb0f1c68b4685a4a47/packages/binding-coap/src/coap-server.ts#L72-L74

Reply:
yes exactly, can you add that prefix, please? :)

Reply:
Aah okay, sure! I thought you mean there is some kind of convention for the way we handle callback errors.

Will be fixed now 😊

Reply:
Done ✅ Is this sufficient?

Reply:
maybe it can be more verbose mentioning the TS file causing the issue

console.error("[binding-coap]", `CoapServer for port ${this.port} failed on observe with: ${err.message}`);

Reply:
I've fixed it. But a short question - won't there be a problem with referencing `this`?

Reply:
I don't think so. It is used elsewhere in the file also. Or do I miss anything?

Reply:
> I've fixed it. But a short question - won't there be a problem with referencing this?

Actually, he might be right, cause the whole code is wrapped inside `function in this case if I recall correctly this points to the `function` object. @fatadel can you do a couple of tests with a small script reproducing the situation? Then we can merge this! 😃

Reply:
while I was here I did the test myself 😄 :
```js
class Test {
    constructor() {
        this.title = "Test";
    }
    testMeFunction(){
        setInterval( function () {
            console.log("function", this.title);
        }, 1000);
    }

    testMeArrow() {
        setInterval( () => {
            console.log("arrow",this.title);
        }, 1000);
    }
}

new Test().testMeFunction();
new Test().testMeArrow();

```
output: 
```
function undefined
arrow Test
function undefined
arrow Test
function undefined

arrow Test
```

Consequently, I'd ask @fatadel to change `function` into an arrow function :)

Reply:
Note: I guess we need to check other places than also...

Reply:
@relu91, done!
Should I also check for `this` inside functions in other places or let's leave it for a separate PR?